addons/stock/models/stock_move_line.py

```diff
@@ -675,20 +675,64 @@ def _get_aggregated_product_quantities(self, **kwargs):
         returns: dictionary {product_id+name+description+uom: {product, name, description, qty_done, product_uom}, ...}
         """
         aggregated_move_lines = {}
-        for move_line in self:
-            name = move_line.product_id.display_name
-            description = move_line.move_id.description_picking
-            if description == name or description == move_line.product_id.name:
+
+        def get_line_key(move, uom=False):
```

Comment:
You should merge this with the `get_aggregated_properties` and call only `get_aggregated_properties` for the `following_move_lines` filtered : 
```python
def get_aggregated_properties(move_line=False, move=False):
            move = move or move_line.move_id
            product = move.product_id
            description = move.description_picking
            uom = move_line and move_line.product_uom_id or move.product_uom
            name = move.product_id.display_name
            if description == name or description == move.product_id.name:
                description = False
            line_key = f'{product.id}_{product.display_name}_{description or ""}_{uom.id}'
            return (name, description, uom, line_key)
```
...
```
...
lambda ml: get_aggregated_properties(move_line=ml)[-1] == line_key
...
```
It should be more readable.

Reply:
Done, but I placed `line_key` at the first place when returned from `get_aggregated_properties`, so we can use the index `0` instead of `-1` in the filtered's lambda (idk, seems more easy/natural).